### ISSUE TITLE
[torch.jit.script] Support tensor indexing after an Ellipsis

### ISSUE THREAD

**Issue by alanhdu** [pytorch/pytorch#34837]
## 🚀 Feature

Support Tensor indexing after an ellipsis: right now the following code:

```python
@torch.jit.script
def zero_diag(inputs: torch.Tensor) -> torch.Tensor:
    dim = inputs.shape[-1]
    inputs[..., torch.range(0, dim), torch.range(0, dim)] = 0
```

fails with:

```
Ellipses followed by tensor indexing is currently not supported:
  File "<ipython-input-105-deac139d15e2>", line 4
def zero_diag(inputs: torch.Tensor) -> torch.Tensor:
    dim = inputs.shape[-1]
    inputs[..., torch.range(0, dim), torch.range(0, dim)] = 0
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
```

## Motivation

For our application, we would like to manipulate the diagonal of a batch of matrices, which is most easily expressed by manipulating `inputs[..., range(N), range(N)]`. Unfortunately, Torch does not currently support this when using `torch.jit.script`.

**Comment by ailzhang** [comment]
Thanks @alanhdu for the feature request! @driazati IIRC you have a list of language features we'll be adding, does this fall into our plan?

**Comment by vfdev-5** [comment]
Various padding mode of tensor can be also implemented with tensor indexing. And it would be very nice to have jit support for that too.

**Comment by Abhijit-2592** [comment]
Any update on this?

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
